traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/CatalogQuery.java

```diff
@@ -18,7 +19,13 @@ public CatalogQuery(List<Flight> flights, List<Product> products) {
         this.products = products;
     }
 
-    public CatalogQuery() { this(new ArrayList<>(), new ArrayList<>()); }
+    public CatalogQuery(String city) {
+        this.city = city;
```

Comment:
router does not handle null flights and products. fix this. either set both to empty arraylist or fix router to hande it. i think its betetr to fix router